plugins/imjournal/imjournal.c

```diff
@@ -472,12 +496,28 @@ pollJournal(void)
 		}
 	}
 
-	assert(r == 1);
+	jr = sd_journal_process(j);
 
-	r = sd_journal_process(j);
-	if (r < 0) {
-		char errStr[256];
+	if (pr == 1 && jr == SD_JOURNAL_INVALIDATE) {
+		/* do not persist stateFile sd_journal_get_cursor will fail! */
+		char* tmp = cs.stateFile;
+		cs.stateFile = NULL;
+		closeJournal(j);
+		cs.stateFile = tmp;
+
+		iRet = openJournal(j);
+		if (iRet != RS_RET_OK) {
+			char errStr[256];
+			rs_strerror_r(errno, errStr, sizeof(errStr));
+			errmsg.LogError(0, RS_RET_IO_ERROR,
```

Comment:
just a small way to improve: we made ```LogError``` available without the need to prefix ```errmsg.``` - it's a little bit more efficient. Of course, all is well, but the new method is the preferred one.